docs/milestones.md

```diff
@@ -0,0 +1,16 @@
+# Milestones
+
+## [M0] - Modelo del Problema
+Entregable: Modelo que represente los elementos clave del problema financiero del estudiante, contenidos en la historia de usuario HU1.  
+Viabilidad: Será viable cuando refleje de manera precisa las entidades clave del problema y las relaciones entre ellas. Otro desarrollador debe poder entender el modelo y continuar con el desarrollo sin ambigüedades.  
+
+## [M1] - Control del Presupuesto
+Entregable: Código que, partiendo del modelo definido en M0, permita el registro y procesamiento de los gastos, proporcionando predicciones sobre los futuros gastos variables.  
```

Comment:
Ya te advertí en el objetivo 0 de que tenías que aclarar muy bien lo que querías hacer para que se pudiera seguir adelante. Sin embargo, no sólo no lo aclaras, sino que pones aquí cosas *excluidas explícitamente en el objetivo0. De [los errores frecuentes del objetivo 0](https://github.com/JJ/IV-/blob/master/errores/objetivo-0.md):

> Los deseos del cliente son órdenes, y eso significa que si el cliente quiere que vuestra aplicación haga algo, tendréis que testear que efectivamente lo hace. Si el problema es maximizar ventas, tendréis que examinar cuales son las ventas con el método actual (o aleatorio, lo que sea) y después con vuestro algoritmo, y ver que es superior estricto. Si queréis predecir, tendréis que programar vosotros un algoritmo que prediga y comprobar con datos que efectivamente lo hace.

¿Vas a hacer un algoritmo que prediga? ¿Vas a implementar ARIMA desde cero? ¿Tienes un conjunto de datos *reales* suficientes para hacer esta predicción?

Por otro lado, ya te lo he enlazado alguna vez, pero quizás sea conveniente hacerlo de nuevo:

> El exceso de detalles hace que se fuerce a soluciones determinadas, y hace que haya que cambiar una cantidad excesiva de cosas. Los milestones son agrupaciones de productos de complejidad creciente, que creen diferentes niveles de abstracción a la hora de trabajar con el problema.

¿Te parece que "predicción" sea un detalle como los que describen aquí?

Por otro lado, también se ha dicho varias veces que hay que poner explícitamente las HUs. Aquí haces referencia al M0 (como si se pudiera hacer de otra forma), que sólo menciona la HU, pero no estaría de más que lo dijeras explícitamente.

Reply:
Dado que el usuario actualmente registra sus gastos en un documento de texto, la idea es que los incluya en el PMV para poder tener esos datos "históricos" y se puedan realizar predicciones con datos "reales", permitiendo añadir el componente estacional como ya me indicó para gastos como agua o electricidad.

Reply:
Respecto a cómo será la predicción. No lo sé. Supongo que la metodoloǵia de desarrollo ágil me hará descubrirlo. Sería un exceso de detalles decir qué se va a hacer exactamente que impediría cualquier proceso de análisis en el desarrollo.

Reply:
> Dado que el usuario actualmente registra sus gastos en un documento de texto, la idea es que los incluya en el PMV para poder tener esos datos "históricos" y se puedan realizar predicciones con datos "reales", permitiendo añadir el componente estacional como ya me indicó para gastos como agua o electricidad.

¿Cómo registra esos gastos? ¿Es el sistema el que va a inferir si son fijos o variables? ¿Qué es lo que va a registrar, de forma precisa? ¿Está en el user journey? ¿Si está, por qué no lo enlazas desde aquí o lo incluyes?

Finalmente, ¿entiendes que no se puede "predecir" nada dentro de esta asignatura?

Reply:
> Respecto a cómo será la predicción. No lo sé. Supongo que la metodoloǵia de desarrollo ágil me hará descubrirlo. Sería un exceso de detalles decir qué se va a hacer exactamente que impediría cualquier proceso de análisis en el desarrollo.

No, la metodología de desarrollo ágil no es magia, y especialmente no es magia que convierta una operación de suma resta trivial en una operación compleja, ni una operación compleja (la programación de un algoritmo de predicción, que necesita entrenamiento y luego un test que incluya posibles márgenes de error) en una operación simple. Sin embargo, todo esto es irrelevante. Lo relevante es que *en el objetivo 0 se excluye explícitamente*, por estas razones y por otras, y además te lo acabo de repetir aquí. ¿Sería un gran problema tratar de ceñirse a problemas que sean aceptables para esta asignatura?

Reply:
Respecto a que si predicción es un detalle que conduce a una solución determinada, la verdad es que pensaba que no, pues daba total libertad al desarrollador sobre cómo elaborar la predicción, pero lo cambiaré.

Reply:
Finalmente, respecto a la referencia  a la historia de usuario, anteriormente lo indicaba y me comentó que estaba mal, por lo que borré la referencia. No entiendo por qué es un error indicar las historias de usuario asociadas a un milestone y también es un error no indicarlas.

Reply:
El usuario registra los datos de forma informal, se en un documento de texto, a mano o de cualquier forma, como indica el User Journey asociado a la HU. Registra el tipo de gasto. Creía que se podía predecir, ya que me hizo reformular el problema, después de aprobarme el objetivo 0, al no encontrarle lógica de negocio válida. A raíz de este comentario (https://github.com/juanmaaf/MoneyController/pull/1#issuecomment-2394355032) entendí que debía elaborar una predición, cálculo a partir existiese un procesamiento de los datos históricos registrados, apartir de la que podría decirle al usuario cúanto podría gastar un fin de semana determinado. Ese iba a ser mi Milestone 2, pero habría que empezar por algo más básico, algo mínimo. Por ello, decidí establecer como M1, una operación similar, manteniendo la misma lógica de negocio. En base a gastos variables pasados, predecir los futuros teniendo en cuenta datos históricos y componentes de tipo estacional. A partir de este control del presupuesto (gestionando gastos fijos y variables) se podría pensar en ahorro y futuros gastos. Pero a partir del objetivo 6 ya he visto que no se sigue trabajando en el desarrollo de milestones y que únicamente se trabaja en 1. No sé cuándo trabajarán en sus milestones los compañeros que tienen el aprobado con 4 milestones.

Reply:
Creía que estaba suficientemente claro que poner sin ningún contexto "historias de usuario asociadas" y poner a bulto dos o tres HUs sin hacer ninguna referencia a las mismas en la descripción del producto era suficientemente autoexplicativo. Pero por si no lo fuera, creo que en el fichero de errores frecuentes dice:

> Aunque los milestones en principio no tienen por qué tener información sobre las historias de usuario, si hay varias historias de usuario hablar de "modelo del problema" o de "lógica de negocio" no es suficiente, porque las dos historias de usuario (o las 5) pueden tener diferentes aspectos del problema, o incluso ser problemas diferentes. En todo caso, siempre es conveniente ser preciso en el lenguaje, porque en el momento que se vaya a trabajar con un milestone determinado puede haber más historias de usuario, o menos, así que aclarar a qué parte del problema del cliente se circunscribe siempre es conveniente.

Eso te lo he idnicado [aquí](https://github.com/juanmaaf/MoneyController/pull/6#discussion_r1795147420) la semana pasada, [aquí](https://github.com/juanmaaf/MoneyController/pull/6#pullrequestreview-2367248392) hace dos días y ahora te lo he vuelto a indicar. ¿No lo entiendes porque no has leído esos comentarios y te has limitado a modificar la línea para eliminar el comentario o porque realmente lo que digo ahí no está suficientemente claro?

Reply:
> El usuario registra los datos de forma informal, se en un documento de texto, a mano o de cualquier forma, como indica el User Journey asociado a la HU. Registra el tipo de gasto. Creía que se podía predecir, ya que me 

No hay ningún user journey asociado ni enlazado a ninguna HU. ¿Es muy complicado enlazar el documento correspondiente?

Pero es que si miramos el UJ 1, dice 

> toma nota de sus gastos fijos,

>  Escribe estos gastos a medida que los recuerda, sin un formato o estructura estricta.

¿Te parece que con esto se puede poner a trabajar alguien?

Si no es para ponerse a trabajar, ¿por qué has hecho esta UJ? ¿Porque te lo he pedido? ¿Sin entender por qué?

Si no entendéis por qué hacéis algo es mejor que no lo hagáis. No lo vais a hacer bien, ni va a servir a ningún propósito. En este caso, sigue sin entenderse cuál es el problema ni qué va a hacer la persona para modelizarlo, porque "escribir los gastos a medida que los recuerda" ¿te parece que se puede empezar a modelizar? ¿Será capaz la persona que trabaje en este repo siquiera de encontrar este UJ, puesto que no está enlazado desde ningún lado?

Reply:
> Entendía que con poner historias de usuario asociadas se entiende que la resolución del milestone conlleva la resolución del problema formulado en las HU mencionadas. Si es mejor hacer referencia en el Entregable o la Viabilidad, lo haré, pero no entiendo la diferencia.

Si entendiste que esa no era la forma de referenciarlas, ¿por qué las eliminaste totalmente en vez de referenciarlas desde cualquier otro sitio?

¿Te parece de todas formas que explicarme lo que entendiste contribuye a que tú avances en el objetivo o a que yo emplee mi tiempo de forma óptima para añadir valor a los estudiantes, a ti y al resto de la clase?

Reply:
Está enlazado desde el README.

Reply:
Reformularé los UJ, permitiendo que alguien pueda modelizar.